### ISSUE TITLE
Timeout Control, Failing Test t/12-exceptions-timeout.t

### ISSUE THREAD

**Issue by wbraswell** [nferraz/perl-openai-api#5]
If I do not have the OPENAI_API_KEY environmental variable set at all, it fails with the following error:

```
$ cpanm -v OpenAI::API
cpanm (App::cpanminus) 1.7042 on perl 5.022001 built for x86_64-linux-gnu-thread-multi
Work directory is /home/FOO/.cpanm/work/1677958512.32205
You have make /usr/bin/make
You have LWP 6.68
You have /bin/tar: tar (GNU tar) 1.28
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Searching OpenAI::API () on cpanmetadb ...
--> Working on OpenAI::API
Fetching http://www.cpan.org/authors/id/N/NF/NFERRAZ/OpenAI-API-0.21.tar.gz ... OK
Unpacking OpenAI-API-0.21.tar.gz
OpenAI-API-0.21/
OpenAI-API-0.21/Changes
OpenAI-API-0.21/MANIFEST
OpenAI-API-0.21/t/
OpenAI-API-0.21/README
OpenAI-API-0.21/META.yml
OpenAI-API-0.21/lib/
OpenAI-API-0.21/Makefile.PL
OpenAI-API-0.21/META.json
OpenAI-API-0.21/lib/OpenAI/
OpenAI-API-0.21/lib/OpenAI/API.pm
OpenAI-API-0.21/lib/OpenAI/API/
OpenAI-API-0.21/lib/OpenAI/API/Request/
OpenAI-API-0.21/lib/OpenAI/API/Request/Embedding.pm
OpenAI-API-0.21/lib/OpenAI/API/Request/Completion.pm
OpenAI-API-0.21/lib/OpenAI/API/Request/Moderation.pm
OpenAI-API-0.21/lib/OpenAI/API/Request/Edit.pm
OpenAI-API-0.21/lib/OpenAI/API/Request/Chat.pm
OpenAI-API-0.21/t/pod.t
OpenAI-API-0.21/t/01-completions.t
OpenAI-API-0.21/t/manifest.t
OpenAI-API-0.21/t/02-edits.t
OpenAI-API-0.21/t/04-chat.t
OpenAI-API-0.21/t/00-load.t
OpenAI-API-0.21/t/12-exceptions-timeout.t
OpenAI-API-0.21/t/03-moderations.t
OpenAI-API-0.21/t/pod-coverage.t
OpenAI-API-0.21/t/11-exceptions-internet.t
OpenAI-API-0.21/t/10-exceptions-validation.t
Entering OpenAI-API-0.21
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.64)
Running Makefile.PL
Configuring OpenAI-API-0.21 ... Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for OpenAI::API
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.64)
Checking if you have Moo 2.005005 ... Yes (2.005005)
Checking if you have Test::Deep 0 ... Yes (1.120)
Checking if you have namespace::clean 0.27 ... Yes (0.27)
Checking if you have Test::RequiresInternet 0 ... Yes (0.05)
Checking if you have Test::Exception 0 ... Yes (0.43)
Checking if you have LWP::Protocol::https 6.10 ... Yes (6.10)
Checking if you have strictures 2.000006 ... Yes (2.000006)
Checking if you have JSON::MaybeXS 1.004002 ... Yes (1.004004)
Checking if you have Test::More 0 ... Yes (1.302191)
Checking if you have LWP::UserAgent 6.47 ... Yes (6.68)
Checking if you have Types::Standard 2.002001 ... Yes (2.002001)
Building and testing OpenAI-API-0.21 ... cp lib/OpenAI/API.pm blib/lib/OpenAI/API.pm
cp lib/OpenAI/API/Request/Embedding.pm blib/lib/OpenAI/API/Request/Embedding.pm
cp lib/OpenAI/API/Request/Chat.pm blib/lib/OpenAI/API/Request/Chat.pm
cp lib/OpenAI/API/Request/Edit.pm blib/lib/OpenAI/API/Request/Edit.pm
cp lib/OpenAI/API/Request/Completion.pm blib/lib/OpenAI/API/Request/Completion.pm
cp lib/OpenAI/API/Request/Moderation.pm blib/lib/OpenAI/API/Request/Moderation.pm
Manifying 6 pod documents
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-load.t ................... 1/? # Testing OpenAI::API 0.21, Perl 5.022001, /usr/bin/perl
t/00-load.t ................... ok   
t/01-completions.t ............ ok   
t/02-edits.t .................. ok   
t/03-moderations.t ............ ok   
t/04-chat.t ................... ok   
t/10-exceptions-validation.t .. ok   
t/11-exceptions-internet.t .... ok   
t/12-exceptions-timeout.t ..... Missing OPENAI_API_KEY at t/12-exceptions-timeout.t line 12.
t/12-exceptions-timeout.t ..... Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 
t/manifest.t .................. skipped: Author tests not required for installation
t/pod-coverage.t .............. skipped: Author tests not required for installation
t/pod.t ....................... skipped: Author tests not required for installation

Test Summary Report
-------------------
t/12-exceptions-timeout.t   (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
Files=11, Tests=15,  6 wallclock secs ( 0.06 usr  0.03 sys +  2.95 cusr  0.54 csys =  3.58 CPU)
Result: FAIL
Failed 1/11 test programs. 0/15 subtests failed.
Makefile:904: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
FAIL
! Installing OpenAI::API failed. See /home/FOO/.cpanm/work/1677958512.32205/build.log for details. Retry with --force to force install it.
```

If I have the OPENAI_API_KEY env var set to a valid value, it fails with the following error:

```
t/00-load.t ................... 1/? # Testing OpenAI::API 0.21, Perl 5.022001, /usr/bin/perl
t/00-load.t ................... ok   
t/01-completions.t ............ ok   
t/02-edits.t .................. ok   
t/03-moderations.t ............ ok   
t/04-chat.t ................... ok   
t/10-exceptions-validation.t .. ok   
t/11-exceptions-internet.t .... ok   
t/12-exceptions-timeout.t ..... 1/? 
#   Failed test 'threw Regexp ((?^:Operation timed out))'
#   at t/12-exceptions-timeout.t line 25.
# expecting: Regexp ((?^:Operation timed out))
# found: Error retrieving 'completions': 500 Can't connect to api.openai.com:443 (Connection timed out) at /home/FOO/.cpanm/work/1677958791.32632/OpenAI-API-0.21/blib/lib/OpenAI/API.pm line 85.
# Looks like you failed 1 test of 1.
t/12-exceptions-timeout.t ..... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
t/manifest.t .................. skipped: Author tests not required for installation
t/pod-coverage.t .............. skipped: Author tests not required for installation
t/pod.t ....................... skipped: Author tests not required for installation

Test Summary Report
-------------------
t/12-exceptions-timeout.t   (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=11, Tests=16, 11 wallclock secs ( 0.07 usr  0.02 sys +  3.65 cusr  0.64 csys =  4.38 CPU)
Result: FAIL
Failed 1/11 test programs. 1/16 subtests failed.
Makefile:904: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
FAIL
! Installing OpenAI::API failed. See /home/FOO/.cpanm/work/1677958791.32632/build.log for details. Retry with --force to force install it.
```

**Comment by nferraz** [comment]
Thanks for the report!

I fixed the test and released version 0.22.

**Comment by wbraswell** [comment]
Yes the tests pass now for `cpanm -v OpenAI::API`, thank you sir!  :-)
